ros/angel_system_nodes/angel_system_nodes/dummy_multi_task_monitor.py

```diff
@@ -0,0 +1,360 @@
+from dataclasses import dataclass, field
+from threading import (
+    Thread,
+)
+from typing import Dict
+from typing import List
+from typing import Set
+import yaml
+
+from builtin_interfaces.msg import Time
+import numpy as np
+from pynput import keyboard
+import rclpy
+from rclpy.node import Node
+
+from angel_msgs.msg import (
+    TaskGraph,
+    TaskUpdate,
+)
+from angel_msgs.srv import QueryTaskGraph
+from angel_utils import declare_and_get_parameters
+from angel_utils.conversion import time_to_int, nano_to_ros_time
+
+
+PARAM_CONFIG_FILE = "config_file"
+PARAM_TASK_STATE_TOPIC = "task_state_topic"
+PARAM_TASK_ERROR_TOPIC = "task_error_topic"
+PARAM_QUERY_TASK_GRAPH_TOPIC = "query_task_graph_topic"
+
+
+@dataclass
+class TaskStateInformation:
+    """
+    Stores relevant task state information for a single task.
+    """
+
+    task_title: str
+    n_steps: int
+
+    # Tracks the current/previous steps
+    # Current step is the most recently completed step the user has finished
+    # as predicted by the HMM.
+    # Previous step is the step before the current step.
+    # Values will be either None or the step string label.
+    current_step = None
+    current_step_id = 0  # HMM ID
+    previous_step = None
+    previous_step_id = 0  # HMM ID
+    # Track the step skips we previously notified the user about to avoid
+    # sending duplicate notifications for the same error.
+    previous_step_skip = None
+    current_step_skip = None
+    # Track the indices of steps we have emitted skip errors for in order
+    # to not repeatedly emit many errors for the same steps.
+    # This may be removed from if step progress regresses.
+    steps_skipped_cache: Set[int] = field(default_factory=set)
+    # Track the latest activity classification end time sent to the HMM
+    # Time is in floating-point seconds up to nanosecond precision.
+    latest_act_classification_end_time = None
```

Comment:
Maybe a silly question: why not just keep this in integer nanoseconds? It seems like there is a bunch of seconds<->nanoseconds conversion happening below. Storing as nanoseconds natively might be simpler in efficiency.

Reply:
This was how `task_monitor_v2` did it. I went ahead and made the change to store in ns here though.